app/components/ChartDataFormatter/index.js

```diff
@@ -4,22 +4,41 @@ import { connect } from 'react-redux';
 import update from 'immutability-helper';
 import NextPrevButton from '../Layout/RebassComponents/NextPrevButton';
 import { locales } from '../../constants/d3Locales';
-import { RECEIVE_CHART_OPTIONS } from '../../constants';
+import { RECEIVE_TICK_FORMAT } from '../../constants';
 import DispatchField from '../lib/DispatchField';
 import {
   defaultTickFormatSettings,
   multiplierOptions,
+  formatScopes,
 } from '../../constants/defaultTickFormatSettings';
+import FormatScopeSelect from './FormatScopeSelect';
+
+const defaultFormatScope = 'all';
+
+/**
+ * For each available format settings scope (axes, tooltip, etc)
+ * merge received tickFormatSettings into default settings.
+ * Look for scope-specific options with fallback
+ * to legacy format that had only one set of options
+ *
+ * @param {Object} settings Received settings object
+ * @return {Object} Options per scope
+ */
+function mergePropsIntoDefaults(settings) {
```

Comment:
This function should ideally be executed in react-redux's mapStateToProps or mergeProps.